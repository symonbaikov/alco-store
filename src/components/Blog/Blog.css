.blog-section {
  margin: 40px 0 0 0;
  padding-bottom: 40px;
}

.blog-title {
  font-size: 2rem;
  font-weight: 500;
  margin-bottom: 32px;
  color: #222;
}

.blog-list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(320px, 1fr));
  gap: 32px 24px;
}

.blog-card {
  display: flex;
  flex-direction: column;
  background: #fff;
  border-radius: 12px;
  box-shadow: 0 2px 8px rgba(0,0,0,0.06);
  text-decoration: none;
  color: inherit;
  overflow: hidden;
  transition: box-shadow 0.2s, transform 0.2s;
  border: 1px solid #eee;
}

.blog-card:hover {
  box-shadow: 0 4px 16px rgba(0,0,0,0.12);
  transform: translateY(-2px) scale(1.01);
}

.blog-card-image-wrapper {
  width: 100%;
  aspect-ratio: 16/9;
  overflow: hidden;
  background: #f5f5f5;
}

.blog-card-image {
  width: 100%;
  height: 100%;
  object-fit: cover;
  display: block;
}

.blog-card-content {
  padding: 18px 20px 16px 20px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.blog-card-date {
  font-size: 13px;
  color: #888;
  margin-bottom: 2px;
}

.blog-card-title {
  font-size: 1.1rem;
  font-weight: 500;
  color: #222;
} 