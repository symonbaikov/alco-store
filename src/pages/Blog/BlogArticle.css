.blog-article-section {
  margin: 40px 0 0 0;
  padding-bottom: 40px;
}

.blog-article-title {
  font-size: 2.2rem;
  font-weight: 600;
  margin-bottom: 12px;
  color: #222;
}

.blog-article-date {
  font-size: 15px;
  color: #888;
  margin-bottom: 24px;
}

.blog-article-image-wrapper {
  width: 100%;
  max-width: 700px;
  margin: 0 auto 32px auto;
  aspect-ratio: 16/9;
  overflow: hidden;
  border-radius: 12px;
  background: #f5f5f5;
}

.blog-article-image {
  width: 100%;
  height: 100%;
  object-fit: cover;
  display: block;
}

.blog-article-content {
  font-size: 1.1rem;
  color: #222;
  line-height: 1.7;
  margin-top: 16px;
  max-width: 700px;
  margin-left: auto;
  margin-right: auto;
} 