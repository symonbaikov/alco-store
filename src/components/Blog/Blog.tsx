import React, { useEffect, useState } from 'react';
import './Blog.css';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

export const Blog: React.FC = () => {
  const { t, i18n } = useTranslation();
  const [blogPosts, setBlogPosts] = useState<any[]>([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    fetch('http://localhost:3001/api/blogs')
      .then(res => res.json())
      .then(data => {
        setBlogPosts(data);
        setLoading(false);
      })
      .catch(() => setLoading(false));
  }, [i18n.language]);

  if (loading) {
    return <div>Загрузка...</div>;
  }

  return (
    <section className="blog-section">
      <div className="container">
        <h2 className="blog-title">{t('blog.title')}</h2>
        <div className="blog-list">
          {blogPosts.map(post => (
            <Link className="blog-card" to={post.link} key={post.id}>
              <div className="blog-card-image-wrapper">
                <img src={post.image} alt={t(`blog.cards.${post.slug}.title`)} className="blog-card-image" />
              </div>
              <div className="blog-card-content">
                <div className="blog-card-date">
                  {t(`blog.cards.${post.slug}.date`).startsWith('blog.cards.')
                    ? new Date(post.date).toLocaleDateString()
                    : t(`blog.cards.${post.slug}.date`)}
                </div>
                <div className="blog-card-title">
                  {t(`blog.cards.${post.slug}.title`).startsWith('blog.cards.')
                    ? post.title
                    : t(`blog.cards.${post.slug}.title`)}
                </div>
              </div>
            </Link>
          ))}
        </div>
      </div>
    </section>
  );
}; 