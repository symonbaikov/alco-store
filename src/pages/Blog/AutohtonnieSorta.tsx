import React from 'react';
import { BlogArticle } from './BlogArticle';
import { useTranslation } from 'react-i18next';

export default function AutohtonnieSorta() {
  const { t } = useTranslation();
  return (
    <BlogArticle
      title={t('blog.autohtonnieSorta.title')}
      date={t('blog.autohtonnieSorta.date')}
      image="/images/zg292c6yc7wsdt7uxyukv8klc6yxhvms.jpg"
    >
      <p>{t('blog.autohtonnieSorta.content1')}</p>
      <p>{t('blog.autohtonnieSorta.content2')}</p>
      <ul>
        <li>{t('blog.autohtonnieSorta.list.0')}</li>
        <li>{t('blog.autohtonnieSorta.list.1')}</li>
        <li>{t('blog.autohtonnieSorta.list.2')}</li>
        <li>{t('blog.autohtonnieSorta.list.3')}</li>
      </ul>
      <p>{t('blog.autohtonnieSorta.content3')}</p>
    </BlogArticle>
  );
} 