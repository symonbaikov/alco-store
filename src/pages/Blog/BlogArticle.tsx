import React from 'react';
import './BlogArticle.css';

interface BlogArticleProps {
  title: string;
  date: string;
  image: string;
  children: React.ReactNode;
}

export const BlogArticle: React.FC<BlogArticleProps> = ({ title, date, image, children }) => (
  <section className="blog-article-section">
    <div className="container">
      <h1 className="blog-article-title">{title}</h1>
      <div className="blog-article-date">{date}</div>
      <div className="blog-article-image-wrapper">
        <img src={image} alt={title} className="blog-article-image" />
      </div>
      <div className="blog-article-content">
        {children}
      </div>
    </div>
  </section>
); 