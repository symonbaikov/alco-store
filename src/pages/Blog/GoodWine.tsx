import React from 'react';
import { BlogArticle } from './BlogArticle';
import { useTranslation } from 'react-i18next';

export default function GoodWine() {
  const { t } = useTranslation();
  return (
    <BlogArticle
      title={t('blog.goodWine.title')}
      date={t('blog.goodWine.date')}
      image="/images/Good-wine.jpg"
    >
      <p>{t('blog.goodWine.content1')}</p>
      <p>{t('blog.goodWine.content2')}</p>
      <ul>
        <li>{t('blog.goodWine.list.0')}</li>
        <li>{t('blog.goodWine.list.1')}</li>
        <li>{t('blog.goodWine.list.2')}</li>
        <li>{t('blog.goodWine.list.3')}</li>
        <li>{t('blog.goodWine.list.4')}</li>
      </ul>
      <p>{t('blog.goodWine.content3')}</p>
    </BlogArticle>
  );
} 