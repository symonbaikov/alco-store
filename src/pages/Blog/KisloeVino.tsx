import { BlogArticle } from './BlogArticle';
import { useTranslation } from 'react-i18next';

export default function KisloeVino() {
  const { t } = useTranslation();
  return (
    <BlogArticle
      title={t('blog.kisloeVino.title')}
      date={t('blog.kisloeVino.date')}
      image="/images/wine.jpg"
    >
      <p>{t('blog.kisloeVino.content1')}</p>
      <ol>
        <li>{t('blog.kisloeVino.list.0')}</li>
        <li>{t('blog.kisloeVino.list.1')}</li>
        <li>{t('blog.kisloeVino.list.2')}</li>
        <li>{t('blog.kisloeVino.list.3')}</li>
        <li>{t('blog.kisloeVino.list.4')}</li>
      </ol>
      <p>{t('blog.kisloeVino.content2')}</p>
    </BlogArticle>
  );
} 