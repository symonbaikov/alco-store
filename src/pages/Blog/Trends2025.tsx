import { BlogArticle } from './BlogArticle';
import { useTranslation } from 'react-i18next';

export default function Trends2025() {
  const { t } = useTranslation();
  return (
    <BlogArticle
      title={t('blog.trends2025.title')}
      date={t('blog.trends2025.date')}
      image="/images/trends-2025.jpg"
    >
      <p>{t('blog.trends2025.content1')}</p>
      <ol>
        <li>{t('blog.trends2025.list.0')}</li>
        <li>{t('blog.trends2025.list.1')}</li>
        <li>{t('blog.trends2025.list.2')}</li>
        <li>{t('blog.trends2025.list.3')}</li>
        <li>{t('blog.trends2025.list.4')}</li>
      </ol>
      <p>{t('blog.trends2025.content2')}</p>
    </BlogArticle>
  );
} 