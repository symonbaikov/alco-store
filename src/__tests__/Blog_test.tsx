import { render, waitFor } from '@testing-library/react';
import { Blog } from '../components/Blog/Blog';
import { MemoryRouter } from 'react-router-dom';

// Мокаем i18next
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => key,
    i18n: {
      language: 'bg',
      changeLanguage: jest.fn()
    }
  })
}));

// Мокаем fetch
beforeAll(() => {
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve([
        {
          id: 1,
          slug: '5-trendov-2025',
          image: '/images/trends-2025.jpg',
          link: '/blog/5-trendov-2025',
          title: '5 Main Trends in the Wine Industry in 2025',
          date: '2025-03-17T00:00:00.000Z'
        }
      ])
    })
  ) as jest.Mock;
});

afterAll(() => {
  jest.resetAllMocks();
});

describe('Blog component', () => {
  it('рендерится корректно (snapshot)', async () => {
    const { asFragment, getByText } = render(
      <MemoryRouter>
        <Blog />
      </MemoryRouter>
    );
    await waitFor(() => {
      // Проверяем, что заголовок появился (ключ перевода)
      expect(getByText('blog.title')).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });
}); 