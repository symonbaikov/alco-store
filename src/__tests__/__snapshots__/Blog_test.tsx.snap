// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog component рендерится корректно (snapshot) 1`] = `
<DocumentFragment>
  <section
    class="blog-section"
  >
    <div
      class="container"
    >
      <h2
        class="blog-title"
      >
        blog.title
      </h2>
      <div
        class="blog-list"
      >
        <a
          class="blog-card"
          data-discover="true"
          href="/blog/5-trendov-2025"
        >
          <div
            class="blog-card-image-wrapper"
          >
            <img
              alt="blog.cards.5-trendov-2025.title"
              class="blog-card-image"
              src="/images/trends-2025.jpg"
            />
          </div>
          <div
            class="blog-card-content"
          >
            <div
              class="blog-card-date"
            >
              3/17/2025
            </div>
            <div
              class="blog-card-title"
            >
              5 Main Trends in the Wine Industry in 2025
            </div>
          </div>
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
